app/models/comment.rb

```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+  validates_associated :user
+  validates_associated :post
+  validates :comment, presence: true, length: { maximum: 250 }
+  validates :username, presence: true, uniqueness: { case_sensitive: true }, length: { maximum: 50 }
+end
```

Comment:
- Since the association between the user and comment and post and comment is established thru lines 2 and 3 respectively, the comment will not be created until a user_id and post_id are provided. Therefore line 4 and 5 are not necessary.
- You can access the username of the comment when you query `comment.user.username`. Therefore, you don't have to create a column of username in comments table as proved on line 7.